src/Storages/MergeTree/MergeTreeBackgroundExecutor.h

```diff
@@ -247,11 +248,13 @@ class MergeTreeBackgroundExecutor final : boost::noncopyable
         String name_,
         size_t threads_count_,
         size_t max_tasks_count_,
-        CurrentMetrics::Metric metric_)
+        CurrentMetrics::Metric metric_,
+        CurrentMetrics::Metric max_tasks_metric_)
         : name(name_)
         , threads_count(threads_count_)
         , max_tasks_count(max_tasks_count_)
         , metric(metric_)
+        , max_tasks_metric(max_tasks_metric_, 2 * max_tasks_count) // active + pending
```

Comment:
Is it always 2? Don't we have a setting for this (`background_merges_mutations_concurrency_ratio`)? Also, looks like `metric` is already multiplied by 2: https://github.com/ClickHouse/ClickHouse/blob/d73de822188bf3e147e359e6ae31d2abafc0e376/src/Interpreters/Context.cpp#L5479

Reply:
It is a different factor. Inside the pool we have active and pending tasks. And capacity set to `max_tasks_count` for both of them.